data/eval-machines.nix

```diff
@@ -101,10 +101,22 @@ in rec {
   };
 
   # Phase 2: build complete machine configurations.
-  machines = { argsFile, buildTargets ? null }:
+  machines = {
+    names ? null,
+    buildTargets ? null,
+    argsJSON ? null,
+  }:
+    assert argsJSON != null -> names == null;
     let
-      fileArgs = builtins.fromJSON (builtins.readFile argsFile);
-      nodes' = filterAttrs (n: v: elem n fileArgs.Names) nodes; in
+      names' = if argsFile != null then
+          (builtins.fromJSON argsJSON).names
+        else
+          names;
```

Comment:
It would be nice to not need this logic. I filed https://github.com/NixOS/nix/issues/6059 to be able to remove it.